README.md

```diff
@@ -21,14 +34,68 @@ $ helm repo add vmpooler-deployment https://puppetlabs.github.io/vmpooler-deploy
 
 ### Adding / updating charts
 
-```bash
-cd docs/
-helm package ../helm-charts/*
-helm repo index --url https://puppetlabs.github.io/vmpooler-deployment/ .
-git add .
-cd ..
-git commit -a
-```
+Make the desired changes to the helm chart in helm-charts/vmpooler and run `./update-chart.sh`.
+
+## Development
+
+Prerequisites:
+
+- [Docker](https://docs.docker.com/engine/install/)
+- [Docker Compose](https://docs.docker.com/compose/install/)
+
+1. Become familiar with the configuration file `docker/vmpooler.yaml` as described in [VMPooler](https://github.com/puppetlabs/vmpooler) Core.
+   - The default configuration file only enables the dummy provider built into [VMPooler](https://github.com/puppetlabs/vmpooler) Core. See each project below for documentation on how to use provider specific options:
+      - [vmpooler-provider-ec2](https://github.com/puppetlabs/vmpooler-provider-ec2)
+      - [vmpooler-provider-gce](https://github.com/puppetlabs/vmpooler-provider-gce)
+      - [vmpooler-provider-vsphere](https://github.com/puppetlabs/vmpooler-provider-vsphere)
+2. Chose a Development method:
+   - Develop via local source:
+      1. Clone all of the known vmpooler projects under a common directory, for example:
+
+           ```bash
+           |-- vmpooler-projects
+           |  |--vmpooler-deployment
+           |  |--vmpooler-provider-ec2
+           |  |--vmpooler-provider-gce
+           |  |--vmpooler-provider-vsphere
+           ```
+
+      2. From the `vmpooler-projects` directory run `docker build -t vmpooler-local -f vmpooler-deployment/docker/Dockerfile.local .`
```

Comment:
```
[com:~/testdir/vmpooler-projects]$ docker build -t vmpooler-local -f vmpooler-deployment/docker/Dockerfile.local .
ERROR: (gcloud.auth.docker-helper) There was a problem refreshing your current auth tokens: ('invalid_grant: Bad Request', {'error': 'invalid_grant', 'error_description': 'Bad Request'})
Please run:

  $ gcloud auth login

to obtain new credentials.

If you have already logged in with a different account:

    $ gcloud config set account ACCOUNT

to select an already authenticated account to use.
```

Reply:
After running `gcloud auth login`:

```
[com:~/testdir/vmpooler-projects]$ docker build -t vmpooler-local -f vmpooler-deployment/docker/Dockerfile.local .
Sending build context to Docker daemon  5.221MB
Step 1/16 : FROM jruby:9.3.6-jdk
 ---> 706cb98073fe
Step 2/16 : LABEL org.opencontainers.image.description "VMPooler plus all known providers"
 ---> Using cache
 ---> fe341449d1f1
Step 3/16 : ENV DEBIAN_FRONTEND=noninteractive
 ---> Using cache
 ---> 4cd0776c040b
Step 4/16 : ENV LOGFILE=/dev/stdout
 ---> Using cache
 ---> 69e5f71b6d56
Step 5/16 : ENV RACK_ENV=production
 ---> Using cache
 ---> 60669f477612
Step 6/16 : RUN apt-get update -qq &&     apt-get upgrade -y &&     apt-get install -y --no-install-recommends make git netbase &&     apt-get clean autoclean &&     apt-get autoremove -y &&     rm -rf /var/lib/apt/lists/*
 ---> Using cache
 ---> 2e25da848515
Step 7/16 : COPY vmpooler /opt/vmpooler
COPY failed: file not found in build context or excluded by .dockerignore: stat vmpooler: file does not exist
```

Reply:
Ah yes, the build context needs to be in the `docker` directory. I'll specify that.

I think the gcloud prompt is unrelated though, there's nothing gcloud specific in there. I guess their helper does that when access expires/is invalid